xarray/core/dask_array_ops.py

```diff
@@ -104,3 +104,49 @@ def func(x, window, axis=-1):
     index = (slice(None),) * axis + (slice(drop_size,
                                            drop_size + orig_shape[axis]), )
     return out[index]
+
+
+def gradient(a, coord, axis, edge_order):
```

Comment:
Do you mean like [this]( http://dask.pydata.org/en/latest/array-api.html#dask.array.gradient )? Requires Dask 0.17.3+ though. So IDK if that works for you or not.

Reply:
Thanks @jakirkahm.
I need to compute the gradient with a non uniform coordinate, but dask Array.gradient only supports uniformly spacing coordinate.

I think this can be implemented using overlap as I do here. Is it in the scope of the dask development?

Reply:
Ah sorry. Should have read more of the code.

Yeah I think that is in scope. Just [not currently supported]( https://github.com/dask/dask/blob/5fa703eaaa6679e452c1dc7cdc8c9ac039994de5/dask/array/routines.py#L437-L438 ) yet.

Could you please open an issue or perhaps PR over on the Dask repo?